apps/frontend/src/modules/workspace/hooks/useWorkspace.tsx

```diff
@@ -47,18 +49,55 @@ export const useWorkspace = () => {
     );
 
     const handleUpdateWorkspace = useCallback(
-        async (organizationInput: OrganizationUpdateInput) => {
+        async (organization: OrganizationUpdateInput | DeepPartial<Organization>, cacheOnly = true) => {
+            if (cacheOnly) {
+                const organizations = {
+                    ...user.organizations,
+                    edges: (user.organizations?.edges ?? []).map((edge) => {
+                        if (edge?.node?.id === (organization as DeepPartial<Organization>).id) {
+                            return {
+                                ...edge,
+                                node: organization as DeepPartial<Organization>,
+                            };
+                        }
+                        return edge;
+                    }),
+                };
+                updateUser(
+                    {
+                        organization: convertToAuthOrganization(organization as DeepPartial<Organization>),
+                        organizations,
+                    },
+                    cacheOnly,
+                );
+                return;
+            }
+
             try {
                 const response = await updateOrganization({
                     variables: {
                         input: {
-                            name: organizationInput.name,
-                            slug: organizationInput.slug,
-                            timezone: organizationInput.timezone,
+                            name: organization.name,
+                            slug: organization.slug,
                         },
                     },
                 });
-                return response?.data?.organizationUpdate;
+                if (response?.data?.organizationUpdate) {
+                    const updatedUser = JSON.parse(JSON.stringify(user)) as DeepPartial<User>;
+                    const currentOrganization = updatedUser.organizations?.edges?.find(
+                        (org) => org?.node?.id === workspace?.id,
+                    );
+                    if (currentOrganization) {
+                        currentOrganization.node = {
+                            ...currentOrganization.node,
+                            name: organization.name || workspace?.name,
+                            slug: organization.slug || workspace?.slug,
+                        };
+                    }
+                    updateUser(updatedUser, true);
+                    toast.success(`Your organization name has been updated`);
```

Comment:
@afolasope UI component in your business logic is a bad design, so you should remove the `toast` here. The issue is that the message won't be the same everywhere, so it's better to localise your toast.